gossip/src/cluster_info.rs

```diff
@@ -899,16 +899,42 @@ impl ClusterInfo {
         }
     }
 
+    pub fn push_epoch_slots(&self, update: &[Slot]) {
+        self.push_epoch_slots_or_restart_last_voted_fork_slots(update, None)
+    }
+
+    pub fn push_restart_last_voted_fork_slots(&self, update: &[Slot], last_vote_bankhash: Hash) {
+        self.push_epoch_slots_or_restart_last_voted_fork_slots(update, Some(last_vote_bankhash))
+    }
+
     // TODO: If two threads call into this function then epoch_slot_index has a
     // race condition and the threads will overwrite each other in crds table.
-    pub fn push_epoch_slots(&self, mut update: &[Slot]) {
+    fn push_epoch_slots_or_restart_last_voted_fork_slots(
+        &self,
+        mut update: &[Slot],
+        last_vote_bankhash: Option<Hash>,
+    ) {
```

Comment:
Why are you mixing this value with `EpochSlots`?
This function has become more convoluted with several `if is_epoch_slot` branches.